tools/setup/emoji/emoji_setup_utils.py

```diff
@@ -254,63 +244,20 @@ def emoji_is_universal(emoji_dict):
+    # type: (Dict[str, Dict[str, Any]]) -> Dict[str, str]
     codepoint_to_name = {}  # type: Dict[str, str]
-    for name in names:
-        codepoint_to_name[unified_reactions_data[name]] = name
+    for emoji_code, name_info in ZULIP_EMOJI_MAP.items():
+        codepoint_to_name[emoji_code] = name_info["canonical_name"]
     return codepoint_to_name
 
-def emoji_can_be_included(emoji_dict, unified_reactions_codepoints):
-    # type: (Dict[str, Any], List[str]) -> bool
-    # This function returns True if an emoji in new(not included in old emoji dataset) and is
-    # safe to be included. Currently emojis which are represented by a sequence of codepoints
-    # or emojis with ZWJ are not to be included until we implement a mechanism for dealing with
-    # their unicode versions.
-    # `:fried_egg:` emoji is banned for now, due to a name collision with `:egg:` emoji in
-    # `unified_reactions.json` dataset, until we completely switch to iamcal dataset.
-    if emoji_dict["short_name"] == "fried_egg":
-        return False
-    codepoint = emoji_dict["unified"].lower()
-    if '-' not in codepoint and emoji_dict["category"] != "Skin Tones" and \
-            emoji_is_universal(emoji_dict) and codepoint not in unified_reactions_codepoints:
-        return True
-    return False
-
-def get_new_emoji_dicts(unified_reactions_data, emoji_data):
-    # type: (Dict[str, str], List[Dict[str, Any]]) -> List[Dict[str, Any]]
-    unified_reactions_codepoints = [unified_reactions_data[name] for name in unified_reactions_data]
-    new_emoji_dicts = []
-    for emoji_dict in emoji_data:
-        if emoji_can_be_included(emoji_dict, unified_reactions_codepoints):
-            new_emoji_dicts.append(emoji_dict)
-    return new_emoji_dicts
-
-def get_extended_names_list(names, new_emoji_dicts):
-    # type: (List[str], List[Dict[str, Any]]) -> List[str]
-    extended_names_list = names[:]
-    for emoji_dict in new_emoji_dicts:
-        extended_names_list.append(emoji_dict["short_name"])
-    return extended_names_list
-
-def get_extended_name_to_codepoint(name_to_codepoint, new_emoji_dicts):
-    # type: (Dict[str, str], List[Dict[str, Any]]) -> Dict[str, str]
-    extended_name_to_codepoint = name_to_codepoint.copy()
-    for emoji_dict in new_emoji_dicts:
-        emoji_name = emoji_dict["short_name"]
-        codepoint = emoji_dict["unified"].lower()
-        extended_name_to_codepoint[emoji_name] = codepoint
-    return extended_name_to_codepoint
-
-def get_extended_codepoint_to_name(codepoint_to_name, new_emoji_dicts):
-    # type: (Dict[str, str], List[Dict[str, Any]]) -> Dict[str, str]
-    extended_codepoint_to_name = codepoint_to_name.copy()
-    for emoji_dict in new_emoji_dicts:
-        emoji_name = emoji_dict["short_name"]
-        codepoint = emoji_dict["unified"].lower()
-        extended_codepoint_to_name[codepoint] = emoji_name
-    return extended_codepoint_to_name
+def generate_name_to_codepoint_map(ZULIP_EMOJI_MAP):
```

Comment:
I think a python convention is that all caps is used only for constants. So this should either be lowercase (zulip_emoji_map, or even, emoji_names), or not take an argument at all.

There are a few more instances of this in this file, I think

Reply:
I want all of these functions to be pure that way they are easy to debug. Changed `ZULIP_EMOJI_MAP -> zulip_emoji_map`.